src/comm/AcceptLimiter.cc

```diff
@@ -24,20 +24,17 @@ Comm::AcceptLimiter::Instance()
 void
 Comm::AcceptLimiter::defer(const Comm::TcpAcceptor::Pointer &afd)
 {
-    ++ (afd->isLimited);
-    debugs(5, 5, afd->conn << " x" << afd->isLimited);
+    debugs(5, 5, "deferring " << afd->conn);
     deferred_.push_back(afd);
 }
 
 void
 Comm::AcceptLimiter::removeDead(const Comm::TcpAcceptor::Pointer &afd)
 {
     uint64_t abandonedClients = 0;
-    for (unsigned int i = 0; i < deferred_.size() && afd->isLimited > 0; ++i) {
-        if (deferred_[i] == afd) {
-            -- deferred_[i]->isLimited;
-            deferred_[i] = NULL; // fast. kick() will skip empty entries later.
-            debugs(5, 5, afd->conn << " x" << afd->isLimited);
+    for (auto it = deferred_.begin(); it != deferred_.end(); ++it) {
+        if (*it == afd) {
+            *it = NULL; // fast. kick() will skip empty entries later.
             ++abandonedClients;
```

Comment:
Since we never defer two pointers to the same TcpAcceptor object, we should debug() and return here. And there will be no need to count the number of matches after that change either -- it will always be one here and zero after the loop. The debug message after the loop should then be adjusted to report the failure to find `afd->conn` in `deferred_.size()`-element queue.